prog/main.go

```diff
@@ -152,28 +154,50 @@ func (c *containerLabelFiltersFlag) String() string {
 }
 
 func (c *containerLabelFiltersFlag) Set(flagValue string) error {
-	newAPITopologyOption, err := c.toAPITopologyOption(flagValue)
+	c.filterIDBase = fmt.Sprintf(c.filterIDBase+"%d", c.filterNumber)
+	newAPITopologyOption, err := c.toAPITopologyOption(flagValue, c.filterIDBase)
+	if err != nil {
+		return err
+	}
+	c.filterNumber++
+
 	c.apiTopologyOptions = append(c.apiTopologyOptions, newAPITopologyOption)
-	return err
+	return nil
 }
 
-func (c *containerLabelFiltersFlag) toAPITopologyOption(flagValue string) (app.APITopologyOption, error) {
-	var indices []int
-	if indices = colonFinder.FindStringIndex(flagValue); len(indices) == 0 {
-		return app.APITopologyOption{}, fmt.Errorf("No unescaped colon found. This is needed to separate the title from the label")
-	} else if len(colonFinder.FindAllStringIndex(flagValue, -1)) > 1 {
+func (c *containerLabelFiltersFlag) toAPITopologyOption(flagValue string, filterID string) (app.APITopologyOption, error) {
+	indexRanges := colonFinder.FindAllStringIndex(flagValue, -1)
+	if len(indexRanges) != 1 {
+		if len(indexRanges) == 0 {
+			return app.APITopologyOption{}, fmt.Errorf("No unescaped colon found. This is needed to separate the title from the label")
+		}
 		return app.APITopologyOption{}, fmt.Errorf("Multiple unescaped colons. Escape colons that are part of the title and label")
 	}
-	var labelKeyValuePair []string
-	containerFilterTitle, containerFilterLabel := unescapeBackslashes.ReplaceAllString(flagValue[0:indices[0]+1], `$1`), unescapeBackslashes.ReplaceAllString(flagValue[indices[1]:], `$1`)
-	if labelKeyValuePair = strings.Split(containerFilterLabel, "="); len(labelKeyValuePair) != 2 {
+	splitIndices := indexRanges[0]
+	titleStringEscaped := flagValue[:splitIndices[0]+1]
+	labelStringEscaped := flagValue[splitIndices[1]:]
+	containerFilterTitle := unescapeBackslashes.ReplaceAllString(titleStringEscaped, `$1`)
+	containerFilterLabel := unescapeBackslashes.ReplaceAllString(labelStringEscaped, `$1`)
+	labelKeyValuePair := strings.Split(containerFilterLabel, "=")
+	if len(labelKeyValuePair) != 2 {
 		return app.APITopologyOption{}, fmt.Errorf("Docker label in the app.container-label-filter(-exclude) flag isn't in the correct key=value format")
 	}
+
 	switch {
```

Comment:
Even better, it would be to create a variable for the filtering function (`render.DoesNotHaveLabel` or `render.HasLabel`) which is the only difference between branches.